internal/dataplane/kong_client_test.go

```diff
@@ -524,129 +538,13 @@ func TestKongClientUpdate_ConfigStatusIsNotified(t *testing.T) {
 	}
 }
 
-func TestKongClientUpdate_PushLastValidConfig(t *testing.T) {
```

Comment:
Why are we removing this test? I think it would be worth actually having a unit test for `KongClient` that would cover the functionality added in this PR.

Reply:
I've added it back. PTAL